src/browser/api_protocol/api_handlers/mesh_middleware.ts

```diff
@@ -137,8 +138,19 @@ function ferryActionMiddleware(msg: MessagePackage, next: () => void) {
     if (isValidUuid && isForwardAction  && isValidIdentity && isRemoteEntity && isLocalAction) {
         try {
             connectionManager.resolveIdentity({uuid})
-            .then((id: any) => {
+            .then((id: IdentityAddress) => {
                 id.runtime.fin.System.executeOnRemote(identity, data)
+                .then(res => {
```

Comment:
I propose:
```
.then(res => {
   if (action === 'get-info' && res && res.data && !res.data.runtime) {
      res.data =  { runtime: { version: id.runtime.portInfo.version } }
   }
   return res;
})
```

Reply:
It's more compact, but the decision to separate the action-matching was deliberate for future readability/expansion.